core/modules/path/path.module

```diff
@@ -178,6 +219,104 @@ function path_form_taxonomy_form_term_alter(&$form, $form_state) {
 }
 
 /**
+ * Implements hook_form_FORM_ID_alter() for taxonomy_form_vocabulary().
+ */
+function path_form_taxonomy_form_vocabulary_alter(&$form, $form_state) {
+  $url_pattern = config_get('path.settings', 'taxonomy_term_pattern');
+  if (isset($form['#vocabulary']->machine_name)) {
+    $specific_url_pattern = config_get('path.settings', 'taxonomy_term_' . $form['#vocabulary']->machine_name . '_pattern');
+    if ($specific_url_pattern != NULL) {
+      $url_pattern = $specific_url_pattern;
+    }
+  }
+
+  $settings = array(
+    'default' => $url_pattern,
+    'description' => t('New pages will have URLs that match a pattern based on wildcards called <em>tokens</em>. For example the URL "tags/environment" could be created using the pattern "[term:vocabulary]/[term:name]" if the vocabulary is named "Tags" and the term is named "Environment".'),
+    'token_types' => array('vocabulary', 'term'),
+  );
+  $form += path_pattern_settings_form($settings);
+  $form['#submit'][] = 'path_taxonomy_form_vocabulary_submit';
+}
+
+/**
+ * Submit handler for taxonomy_form_vocabulary().
+ */
+function path_taxonomy_form_vocabulary_submit($form, &$form_state) {
+  if (isset($form_state['values']['path_settings'])) {
+    $config = config('path.settings');
+    $pattern_type  = 'taxonomy_term_' . $form['#vocabulary']->machine_name . '_pattern';
+    // Don't save the specific pattern if it matches the general pattern.
+    if ($config->get('taxonomy_term_pattern') != $form_state['values']['path_settings']) {
+      $config->set($pattern_type, $form_state['values']['path_settings']);
+    }
+    else {
+      // If it matches the general pattern delete the specific pattern.
+      $config->clear($pattern_type);
+    }
+    $config->save();
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter() for user_admin_settings().
+ */
+function path_form_user_admin_settings_alter(&$form, $form_state) {
+  $settings = array(
+    'default' => config_get('path.settings', 'user_pattern'),
+    'description' => t('New pages will have URLs that match a pattern based on wildcards called <em>tokens</em>. For example the URL "members/jane-doe" could be created using the pattern "members/[user:name]" if the user name is "Jane Doe".'),
+    'token_types' => array('user'),
+  );
+  $form += path_pattern_settings_form($settings);
+  $form['#submit'][] = 'path_user_admin_settings_submit';
+}
+
+/**
+ * Submit handler for user_admin_settings().
+ */
+function path_user_admin_settings_submit($form, &$form_state) {
+  if (isset($form_state['values']['path_settings'])) {
+    config_set('path.settings', 'user_pattern', $form_state['values']['path_settings']);
+  }
+}
+
+/**
+ * Return URL pattern settings form for easy addition into other forms.
+ *
+ * @param $settings
+ *   Array of variabels needed to complete the form, including:
```

Comment:
how do you see things like this?  Magic eyes :) Can you open a new issue? Maybe tag it novice?